src/Elevator.java

```diff
@@ -170,25 +188,49 @@ public void run() {
 						currState = currState.nextState();
 						break;
 					}
-	
+
 					break;
 				}
-	
+
 				case ArriveReqFloor: {
+					try {
+						this.setDoors(true);
+						Thread.sleep(1100);
+						if (!this.isDoorsOpen()) {
+//							currState=TRANSIENTSTATE;
+							System.out.println("Elevator " + this.getElID()+ "in transient fault state");
+							break;
+						}
+					} catch (InterruptedException e) {
+						System.out.println("Door open Timer interrupted. Who did that?");
+					}
+
 					currState = currState.nextState();
 					break;
-	
+
 				}
-	
+
 				case ReqFloorDoorsOpened: {
+					try {
+						this.setDoors(false);
+						Thread.sleep(1100);
+						if (this.isDoorsOpen()) {
+//							currState=TRANSIENTSTATE;
```

Comment:
This line will need to be changed and uncommented when the transient fault state is implemented